### PR TITLE
Precompiler

### PR DESCRIPTION
**Pull request by duanefields**
Precompiler that fills the cache prior to starting the server

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,19 +25,21 @@
...
   "homepage": "https://github.com/Custom-Elements/polymer-serve",
   "dependencies": {
-    "bluebird": "^2.5.3",
+    "bluebird": "^2.9.25",
     "browserify": "10.1.2",
+    "cheerio": "^0.19.0",
     "coffee-script": "^1.8.0",
     "coffeeify": "^1.0.0",
     "colors": "~0.6.2",
+    "cors": "^2.5.2",
```
Comment (wballard) [inline package.json:34]: ? starphleet sets CORS headers

File: src/polymer-serve.litcoffee [hunk 2]
```diff
@@ -12,15 +12,19 @@ server, which will be transpiled into polymer ready browser custom elements.
...
+      --precache         Precompile and cache all resources before starting, implies --cache
       --quiet            SSSHHH! Less logging.
     """
     {docopt} = require 'docopt'
     _ = require 'lodash'
     args = docopt(doc)
-    path = require 'path'
-    fs = require 'fs'
+    Promise = require 'bluebird'
+    Path = Promise.promisifyAll require 'path'
```
Comment (wballard) [inline src/polymer-serve.litcoffee:22]: Huh? Path is just a stringy thing

File: src/polymer-serve.litcoffee [hunk 3]
```diff
@@ -51,9 +55,71 @@ Using cluster to get a faster build -- particularly on the initial request.
...
+      compileMarkdownAsync = require('./markdown-middleware.litcoffee')(args, args.root_directory).compile
+      compileScriptAsync = require('./script-middleware.litcoffee')(args, args.root_directory).compile
+      compileStyleAsync = require('./style-middleware.litcoffee')(args, args.root_directory).compile
 
       app.use express.static(args.root_directory)
-      app.listen port
+
+Optional precache step to precompile and populate the cache, before the server becomes available
+
+      if not args['--precache']
```
Comment (wballard) [inline src/polymer-serve.litcoffee:70]: ONly trouble I'm seeing here is that if we add another file type, we get to go in two places, wheras with the middleware before without caching it was a singular add.

Comment (duanefields) [inline src/polymer-serve.litcoffee:70]: > ONly trouble I'm seeing here is that if we add another file type

I see.....

**Pull request merged**